### ISSUE TITLE
[ic-select] Loading and error state accessibility issues

### ISSUE THREAD

**Issue by lz405** [mi6/ic-ui-kit#4392]
### Summary of the bug 
On an ic-select, the loading state and loading error message aren't read out by a screen reader

### 🪜 How to reproduce 
Tell us the steps to reproduce the problem: 
1. Go to page: [Error/loading storybook](https://mi6.github.io/ic-ui-kit/branches/develop/react/iframe.html?globals=&args=&id=select-single--loading-with-error&viewMode=story)
2. Turn on screen reader
3. Click on: The select dropdown
4. The screen reader does not read out anything related to the loading state, or the error message

### 🖥 📱 Device 
- Type: [e.g. Desktop, mobile] 
- Device: [e.g. iPhone, MacBook, ThinkPad] 
- OS version: [e.g. macOS13, iOS16, Android 13] 
- Browser version: [e.g. Chrome 115, Safari 16] 

### 🧐 Expected behaviour 
The screen reader should give some information regarding the loading state
The screen reader should read out the error message

### 📝 Acceptance Criteria  
If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil. 

*Given* An ic-select is in an error or loading state
*When* A screen reader user navigates onto the ic-select
*Then* The screen reader announces the error message, or the state of loading
